# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1451070448 -3600
#      Fri Dec 25 20:07:28 2015 +0100
# Node ID 8a43428e13b2045b616e91f81e77d76d00ddb155
# Parent  167c536419a7def253d114d7d625d2dca83d864e
fix buttons roles in save-and-run-file dialog box (bug #46753)

* octave-qt-link.cc (do_debug_cd_or_addpath_error): set appropriate roles
  to the buttons of the dialog

diff --git a/libgui/src/octave-qt-link.cc b/libgui/src/octave-qt-link.cc
--- a/libgui/src/octave-qt-link.cc
+++ b/libgui/src/octave-qt-link.cc
@@ -391,24 +391,24 @@ octave_qt_link::do_debug_cd_or_addpath_e
 
   QString cd_txt = tr ("Change Directory");
   QString addpath_txt = tr ("Add Directory to Load Path");
   QString cancel_txt = tr ("Cancel");
 
   QStringList btn;
   QStringList role;
   btn << cd_txt;
-  role << "AcceptRole";
+  role << "YesRole";
   if (addpath_option)
     {
       btn << addpath_txt;
       role << "AcceptRole";
     }
   btn << cancel_txt;
-  role << "AcceptRole";
+  role << "RejectRole";
 
   // Lock mutex before signaling.
   uiwidget_creator.mutex.lock ();
 
   uiwidget_creator.signal_dialog (msg, title, "quest", btn, cancel_txt, role);
 
   // Wait while the user is responding to message box.
   uiwidget_creator.waitcondition.wait (&uiwidget_creator.mutex);
